marketplaceItems/audioFocus/appResources/appData/audioFocus_app.js

```diff
@@ -0,0 +1,459 @@
+        removeSolo(avatarUUID);
+    }
+
+
+    // Handles avatar being solo'd
+    function soloAvatar(avatarUUID) {
+        log("in soloAvatar", avatarUUID);
+
+        log("soloAvatars", soloAvatars);
+
+        if (!isEnabled) {
+            return;
+        }
+
+        var getAvatarClicked = AvatarList.getAvatar(avatarUUID);
+        var displayUsername = getAvatarClicked.sessionDisplayName;
+
+        log("avatar clicked", displayUsername);
+
+        if (soloAvatars[avatarUUID]) {
+            removeAvatarFromList(avatarUUID);
+        } else {
+            addAvatarToList(avatarUUID, displayUsername);
+        }
+    }
+
+
+    // #endregion
+    // *************************************
+    // STOP SOLO FUNCTIONS
+    // *************************************
+
+    // *************************************
+    // START OVERLAY FUNCTIONS
+    // *************************************
+    // #region Overlay
+
+    // Adds a speaker overlay above a user solo'd
+    function addOverlayToUser(uuid) {
+        var user = soloAvatars[uuid];
+        var overlayPosition = AvatarList.getAvatar(uuid).getNeckPosition(); // user.currentPosition
+
+        var WHITE = [255, 255, 255];
+
+        var overlayProperties = {
+            position: Vec3.sum(overlayPosition, [0, 0.75, 0]),
+            dimensions: { x: 0.3, y: 0.3, z: 0.3 },
+            alpha: 1.0,
+            color: WHITE,
+            parentID: uuid,
+            drawInFront: true,
+            url: Script.resolvePath("./resources/images/speaker.png")
+        };
+
+        var overlayID = Overlays.addOverlay("image3d", overlayProperties);
+        user.overlayID = overlayID;
+    }
+
+
+    // Removes the overlay from a uuid
+    function removeOverlay(uuid) {
+        var user = soloAvatars[uuid];
+
+        Overlays.deleteOverlay(user.overlayID);
+
+        user.overlayID = null;
+    }
+
+
+    // Removes all overlays from a user
+    function removeAllOverlays() {
+        // remove previous overlays
+        for (var uuid in soloAvatars) {
+            removeOverlay(uuid);
+        }
+    }
+
+
+    // #endregion
+    // *************************************
+    // STOP OVERLAY FUNCTIONS
+    // *************************************
+
+
+    // *************************************
+    // START TABLET FUNCTIONS
+    // *************************************
+    // #region Tablet
+
+    var BUTTON_NAME = "AUDIO FOCUS";
+    var URL = Script.resolvePath('./resources/audioFocus_ui.html');
+    var appUi = Script.require('appUi');
+
+    var ui = new appUi({
+        buttonName: BUTTON_NAME,
+        home: URL,
+        graphicsDirectory: Script.resolvePath("./icons/tablet-icons/"),
+        onOpened: onOpened,
+        onClosed: onClosed,
+        onMessage: onMessage
+    });
+
+    // function for appUi to call when opened
+    function onOpened() {
+        log("onOpened");
+
+        Controller.mousePressEvent.connect(mousePressEvent);
```

Comment:
i mentioned this in another comment - but consider moving this to inside `enable()`